homeassistant/helpers/condition.py

```diff
@@ -239,6 +242,21 @@ def disabled_condition(
 
             return disabled_condition
 
+    condition: str = config[CONF_CONDITION]
+    factory: Any = None
+    platform = await _async_get_condition_platform(hass, config)
+
+    if platform is None:
+        for fmt in (ASYNC_FROM_CONFIG_FORMAT, FROM_CONFIG_FORMAT):
+            factory = getattr(sys.modules[__name__], fmt.format(condition), None)
+
+            if factory:
+                break
+    else:
+        condition_descriptors = await platform.async_get_conditions(hass)
+        condition_instance = condition_descriptors[condition](hass, config)
+        return await condition_instance.async_condition_from_config()
```

Comment:
```suggestion
    if platform is not None:
        condition_descriptors = await platform.async_get_conditions(hass)
        condition_instance = condition_descriptors[condition](hass, config)
        return await condition_instance.async_condition_from_config()
        
    for fmt in (ASYNC_FROM_CONFIG_FORMAT, FROM_CONFIG_FORMAT):
        factory = getattr(sys.modules[__name__], fmt.format(condition), None)

        if factory:
            break
```
Since we now return when the platform is not None, we can move that into the `if` and do an early return, and leave the rest out since that is what continues